cmd/primateRecon/primateRecon.go

```diff
@@ -5,29 +5,175 @@ package main
+// and gives back the reconstructed base for the hca
+func likelihoodsToBaseUnbiased(likes []float64, probThreshold float64) dna.Base {
+	var total, bestProb float64
+	var i int
+	var answer dna.Base = dna.N
+
+	for i = range likes {
+		total += likes[i]
+	}
+
+	for i = range likes {
+		if likes[i]/total >= probThreshold && likes[i] > bestProb {
+			bestProb = likes[i]
+			answer = dna.Base(i)
+		}
+	}
+	return answer
+}
+
+// likelihoodsToBaseHumanBias takes the un-normalized likelihoods for A, C, G, T as well
+// as the index of the human base (0 for A, 1 for C, etc), and the probability
+// threshold for when we will call it the mle base instead of the human base
+// and gives back the reconstructed base for the hca
+func likelihoodsToBaseHumanBias(likes []float64, humanBase dna.Base, probThreshold float64, nonHumanProbThreshold float64) dna.Base {
+	var total, nonHumanTotal, bestProb float64
+	var i int
+	var answer dna.Base
+
+	if humanBase == dna.Gap {
+		answer = dna.N
+	} else {
+		answer = humanBase
+	}
+
+	for i = range likes {
+		total += likes[i]
+		if i != int(humanBase) {
+			nonHumanTotal += likes[i]
+		}
+	}
+
+	for i = range likes {
+		if likes[i]/total >= probThreshold && likes[i] > bestProb && nonHumanTotal/total >= nonHumanProbThreshold {
+			bestProb = likes[i]
+			answer = dna.Base(i)
+		}
+	}
+	return answer
+}
+
+func baseIsPresent(b dna.Base) bool {
+	if dna.DefineBase(b) || b == dna.N {
+		return true
+	}
+	return false
+}
+
+func hcaIsPresent(human, bonobo, chimp, gorilla, organutan dna.Base) bool {
+	if baseIsPresent(human) && (baseIsPresent(bonobo) || baseIsPresent(chimp)) {
+		return true
+	}
+	if (baseIsPresent(human) || baseIsPresent(bonobo) || baseIsPresent(chimp)) && (baseIsPresent(gorilla) || baseIsPresent(organutan)) {
+		return true
+	}
+	return false
+}
+
+func reconHcaBase(root, humanNode, nodeToRecon *expandedTree.ETree, position int, probThreshold float64, nonHumanProbThreshold float64, humanBias bool) {
+	var likelihoods []float64
+	var nextBase dna.Base
+	reconstruct.SetState(root, position)
+	likelihoods = reconstruct.FixFc(root, nodeToRecon)
+	if humanBias {
+		nextBase = likelihoodsToBaseHumanBias(likelihoods, humanNode.Fasta.Seq[position], probThreshold, nonHumanProbThreshold)
+	} else {
+		nextBase = likelihoodsToBaseUnbiased(likelihoods, probThreshold)
+	}
+	nodeToRecon.Fasta.Seq = append(nodeToRecon.Fasta.Seq, nextBase)
+}
+
+func primateReconMle(inFastaFilename string, inTreeFilename string, humanBias bool, probThreshold float64, nonHumanProbThreshold float64, outputFastaFilename string) {
+	var tree, humanNode, humanAltNode, bonoboNode, chimpNode, gorillaNode, orangutanNode, hcaNode *expandedTree.ETree
+	var err error
+	var i int
+
+	tree, err = expandedTree.ReadTree(inTreeFilename, inFastaFilename)
+	exception.FatalOnErr(err)
+
+	// roll call to make sure everyone is here and will need them later
+	humanNode = expandedTree.FindNodeName(tree, "hg38")
```

Comment:
Is this program specific to certain reference genome assemblies (e.g. hg38, gorGor5)? To me it looks like the user needs to provide inputs where the program can find the specific genome assemblies. Does the specific genome assembly requirement need to be specified in either usage or the documentation?